lib/mergeusersearch.php

```diff
@@ -107,9 +113,15 @@ public function search_users($input, $searchfield){
 
                 break;
             default: // search on all fields by default
+                if (!ctype_digit($input)) {
```

Comment:
Idem as before. Use `$DB->get_in_or_equal()` API to build this part of the SQL and the corresponding array of parameters.

Reply:
Again, I don't think `$DB->get_in_or_equal()` helps us here, I've reworked the use of `ctype_digit()` to `is_string($input) && !ctype_digit($input)`.

Reply:
Thinking about this again my original approach could be improved by using `$DB->sql_cast_to_char()` so instead of `WHERE id = 'abc'` which gets an error in PostgreSQL, we now have `WHERE CAST(id AS VARCHAR) = 'abc'` which avoids this error. So I've added a further commit (09994f2) with this change.

(All CI tests have failed but I don't think that's to do with my commit).